zc.recipe.egg_/src/zc/recipe/egg/egg.py

```diff
@@ -57,40 +57,34 @@ def working_set(self, extra=()):
         This is intended for reuse by similar recipes.
         """
         options = self.options
-        b_options = self.buildout['buildout']
+        buildout_section = self.buildout['buildout']
 
         # Backward compat. :(
         options['executable'] = sys.executable
 
-        distributions = [
-            r.strip()
-            for r in options.get('eggs', self.name).split('\n')
-            if r.strip()]
-        orig_distributions = distributions[:]
-        distributions.extend(extra)
-
-        if self.buildout['buildout'].get('offline') == 'true':
-            ws = zc.buildout.easy_install.working_set(
-                distributions,
-                [options['develop-eggs-directory'], options['eggs-directory']]
-                )
-        else:
-            ws = zc.buildout.easy_install.install(
-                distributions, options['eggs-directory'],
-                links=self.links,
-                index=self.index,
-                path=[options['develop-eggs-directory']],
-                newest=self.buildout['buildout'].get('newest') == 'true',
-                allow_hosts=self.allow_hosts)
-
-        return orig_distributions, ws
+        return working_set(
+            distributions=[
+                r.strip()
+                for r in options.get('eggs', self.name).split('\n')
+                if r.strip()
+            ],
+            develop_eggs_dir=options['develop-eggs-directory'],
+            eggs_dir=options['eggs-directory'],
+            offline=(buildout_section.get('offline') == 'true'),
+            newest=(buildout_section.get('newest') == 'true'),
+            links=self.links,
+            index=self.index,
+            allow_hosts=self.allow_hosts,
+            extra=extra,
+            )
 
     def install(self):
         reqs, ws = self.working_set()
         return ()
 
     update = install
 
+
```

Comment:
> I agree with the suggestions. But how can I do that? My git-fu is weak. Perhaps close this PR and do a new one?

It's not necessary to close this PR. As long as this PR is open, whatever is on `rafaelbco:master` will be what this PR represents, but you'll need to do `git rebase -i upstream/master` to reorder your history and squash commits, then `git push -f` to force push the rewritten branch on top of the one in your github repository.

This assumes you have a git "remote" called `upstream` pointing to the original buildout repo.

Check some tutorials on `git interactive rebase` or `git rewrite history`.

In any case it's better to do the history rewriting as the last step, after replying to all change requests with pure additional commits.

But in the end, if after making all changes you're still uncomfortable doing git history rewriting, just give me permission to your repo and I'll rewrite the history for you. The easiest way to do that is by [allowing changes to a pull request](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/).